### PR TITLE
BUG:subsidy risk of coinbase outputs

### PR DESCRIPTION
**Pull request by lochjin**

### CODE REVIEW

File: core/blockchain/validate.go [hunk 1]
```diff
@@ -355,24 +355,35 @@ func CheckTransactionSanity(tx *types.Transaction, params *params.Params) error
...
-			curPkScriptStr := hex.EncodeToString(tx.TxOut[1].PkScript)
-			if orgPkScriptStr != curPkScriptStr {
-				str := fmt.Sprintf("coinbase transaction for block pays to %s, but it is %s",
-					orgPkScriptStr, curPkScriptStr)
-				return ruleError(ErrBadCoinbaseValue, str)
-			}
-
-		} else if len(tx.TxOut) >= 3 {
+		}
+		if len(tx.TxOut) >= 3 {
```
Comment (dindinw) [inline core/blockchain/validate.go]: The `if` statement results differently than the original `else if` statement, please do a double review of the change here.

File: core/blockchain/validate.go [hunk 2]
```diff
@@ -533,20 +544,20 @@ func (b *BlockChain) checkBlockSubsidy(block *types.SerializedBlock, totalFee in
 	// check subsidy
 	transactions := block.Transactions()
 	subsidy := b.subsidyCache.CalcBlockSubsidy(int64(blocks))
-	workAmountOut := int64(transactions[0].Tx.TxOut[0].Amount)
-
-	hasTax := false
-	if b.params.BlockTaxProportion > 0 &&
-		len(b.params.OrganizationPkScript) > 0 {
-		hasTax = true
+	workAmountOut := int64(0)
```
Comment (dindinw) [inline core/blockchain/validate.go]: The logic change here requires the unit-test coverage to convince the correctness. 
It's difficult to understand the change from `tx[0].output[0].amount` to `tx[0].output[2:].amount`

Comment (lochjin) [reply core/blockchain/validate.go]: https://github.com/Qitmeer/qitmeer/pull/261/commits/0b232fced328fbc073dfb63676d3392ceb41e2c6

File: core/blockchain/validate.go (hunk 1 above)
Comment (lochjin) [reply core/blockchain/validate.go]: 👌

**Pull request merged**